### PR TITLE
fix: [P4PU-665] Managing the empty state for filtered paid notices

### PR DESCRIPTION
**Pull request by stratoivandiluccio**
#### List of Changes
- QueryLoader component now accepts a new  boolean 'loading' prop to manage the loading state from 'outside' 
- The getNoticesList hook (by the way we should change the name into 'useGetNoticesList') now accepts an array of dependencies to manage updates
- A new 'NoData' component is created to be used when no filtered paid notices are returned
- useNormalizedNoticesList hook and related unit tests deleted

#### Motivation and Context
The edge case was not managed and when no filtered paid notices were returned the users were stuck on the page

#### How Has This Been Tested?
Tested automatically updating all unit tests involved and manually testing the local FE

#### Screenshots (if appropriate):

image

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/QueryLoader/index.tsx [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
-      setTimeout(() => setLeast(0), props.atLeast);
+      setTimeout(() => setAtLeast(0), props.atLeast);
     }
   }, [isFetching, props.atLeast]);
 
   const loader = loaderComponent || <CircularProgress />;
-  return <>{isFetching + atLeast ? loader : children}</>;
+  return (
+    <>
+      {loading !== undefined
```
Comment (mmoio) [inline src/components/QueryLoader/index.tsx:35]: We could add a local variable like:
```js
const isLoading = loading?? isFetching || atLeast > 0
```
and change this ternary with
```js
 isLoading? loader:children
 ```

**Review by mmoio — CHANGES_REQUESTED** [review]

File: src/components/QueryLoader/index.tsx (hunk 1 above)
Comment (stratoivandiluccio) [reply src/components/QueryLoader/index.tsx:35]: What you are suggesting to me implies a different logic. 
if I apply the suggestions you give me the functioning is no longer as desired

**Review by vito80ba — APPROVED** [review]

**Review by mmoio — APPROVED** [review]
